[bug-fix] riff: use channel information for adpcm

Description: mono channel adpcm files generate Pop-noise or skipped play.
diff -Naur 1.16.2-r0/gst-plugins-base-1.16.2/gst-libs/gst/riff/riff-media.c after/gst-libs/gst/riff/riff-media.c
--- 1.16.2-r0/gst-plugins-base-1.16.2/gst-libs/gst/riff/riff-media.c	2021-08-09 14:50:36.645695952 +0900
+++ after/gst-libs/gst/riff/riff-media.c	2021-08-17 18:07:57.593781179 +0900
@@ -1307,7 +1307,7 @@
          * would probably confuse timing */
         strf->av_bps = 0;
         if (strf->channels != 0 && strf->rate != 0 && strf->blockalign != 0) {
-          int spb = ((strf->blockalign - strf->channels * 7) / 2) * 2;
+          int spb = ((strf->blockalign - strf->channels * 7) / strf->channels) * 2;
           strf->av_bps =
               gst_util_uint64_scale_int (strf->rate, strf->blockalign, spb);
           GST_DEBUG ("fixing av_bps to calculated value %d of MS ADPCM",
@@ -1429,7 +1429,7 @@
          * as this would probably confuse timing */
         strf->av_bps = 0;
         if (strf->channels != 0 && strf->rate != 0 && strf->blockalign != 0) {
-          int spb = ((strf->blockalign - strf->channels * 4) / 2) * 2;
+          int spb = ((strf->blockalign - strf->channels * 4) / strf->channels) * 2;
           strf->av_bps =
               gst_util_uint64_scale_int (strf->rate, strf->blockalign, spb);
           GST_DEBUG ("fixing av_bps to calculated value %d of IMA DVI ADPCM",
